### PR TITLE
Set up association

### PR DESCRIPTION
**Pull request by Hassaanjbaig-code**
## In this pull request 

> Create a Class  `classname` and add all code [according to the requirement](https://github.com/microverseinc/curriculum-ruby/blob/d75bb1fd092519fe76f12753a0a8684f44d31110/oop/school_library_associations.md?plain=1#L24-L26)

> Create a `has-many`/`belongs-to` relationship between `Classroom` and `Student` and [implement the requirement ](https://github.com/microverseinc/curriculum-ruby/blob/d75bb1fd092519fe76f12753a0a8684f44d31110/oop/school_library_associations.md?plain=1#LL27C3-L31C105)

> Create a class `Book`  and add all code [according to the requirement](https://github.com/microverseinc/curriculum-ruby/blob/d75bb1fd092519fe76f12753a0a8684f44d31110/oop/school_library_associations.md?plain=1#LL32C3-L34C81)

> Create a class `Rental` and add all code [according to the requirement](https://github.com/microverseinc/curriculum-ruby/blob/d75bb1fd092519fe76f12753a0a8684f44d31110/oop/school_library_associations.md?plain=1#LL35C3-L37C68)

> Create a `many to many` relation and add all code [according to the requirement](https://github.com/microverseinc/curriculum-ruby/blob/d75bb1fd092519fe76f12753a0a8684f44d31110/oop/school_library_associations.md?plain=1#LL38C3-L43C77)

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,12 +1,26 @@
-class Person
+require_relative './nameable'
+require_relative './capitalizedecorator'
+require_relative './trimmer_decorator'
+
+class Person < Nameable
   attr_accessor :name, :age
```
Comment (Ghiftee) [inline person.rb]: - [x] For rentals to be both accessed and edited by a person, it needs to be given both getters and setters methods like name and age, kindly add ```rentals``` to ```attr_accessor```

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @Hassaanjbaig-code 👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ No linter errors
✔️ README is professional
✔️ Correct associations are set up

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._
- [ ] Your MIT license in your README file is broken, kindly fix it as it adds to the professionalism of your project documentation.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: person.rb (hunk 1 above)
Comment (Hassaanjbaig-code) [reply person.rb]: @Ghiftee  Thank you for review

**Review by BregornOriginal — APPROVED** [review]
## Hi @Hassaanjbaig-code 

## Your project is complete!  There is nothing else to say other than... it's time to merge it 

## Congratulations! 

## Highlights

✅ Linters are passing
✅ Good work implementing changes requested by the previous reviewer

## Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.
Cheers and Happy coding!:aplauso::aplauso::aplauso:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, remember to tag me in your question so I can receive the notification.
------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)[.](https://airtable.com/shrQAqnBwek5a0O0s)._)

**Pull request merged**